corehq/apps/locations/models.py

```diff
@@ -282,7 +282,7 @@ def couch_locations(self, wrapped=True):
 
     def accessible_to_user(self, domain, user):
         if user.has_permission(domain, 'access_all_locations'):
-            return self.all()
+            return self.filter(domain=domain)
```

Comment:
`user.get_location_ids(domain)` should only return ids of locations in that domain (otherwise there's some other big problem somewhere), but it couldn't hurt as an additional safety measure.